trpcultivate_phenotypes/src/Plugin/Validators/ValidDelimitedFile.php

```diff
@@ -0,0 +1,211 @@
+          // Use the strict flag of the validator columns configuration to compare the columns returned by
+          // split method and the configured columns.
+          
+          // A strict set to True means exact match whereas False requires at least the configured columns.
+          $delimiters_checked = [];
+          foreach($delimiters_used as $delimiter) {
+            $columns = TripalCultivatePhenotypesValidatorBase::splitRowIntoColumns($raw_row, $input_file_mime_type);
+
+            if ($expected_columns['strict']) {
+              // A strict comparison - exact match only.
+              if (count($columns) != $expected_columns['number_of_columns']) {
+                array_push($delimiters_checked, $delimiter);
+              }
+            }
+            else {
+              // Not a strict comparison - at least x number of columns.
+              if (count($columns) < $expected_columns['number_of_columns']) {
+                array_push($delimiters_checked, $delimiter);
+              }
+            }
+          }
+
+          // If all delimiters failed the checks, then the line failed due to none of the
+          // delimiter was able to split the line into the expected number of columns.
+          if ($delimiters_used == $delimiters_checked) {
+            $case = 'Raw row is not delimited';
+            $valid = FALSE;
+            $failed_items = ['raw_row' => $raw_row];
+          }
+        }
+        else {
+          // Neither of the supported delimiters for the file was detected in the raw row.
+          $case = 'None of the delimiters supported by the file type was used';
+          $valid = FALSE;
+          $failed_items = ['raw_row' => $raw_row];
+        }
+      }      
+    }
+    else {
+      // The line provided is an empty string.
+      $case = 'Raw row is empty';
+      $valid = FALSE;
+      $failed_items = ['raw_row' => 'is an empty string value'];
+    }
+    
+    return [
+      'case' => $case,
+      'valid' => $valid,
+      'failedItems' => $failed_items
+    ];
+  }
+
+  /**
+   * Set a number of required columns.
+   * 
+   * @param integer $number_of_columns
+   *   An integer value greater than zero.  
+   * @param bool $strict
+   *   This will indicate how to compare the the number of columns set against
+   *   the number of columns returned by the split line method.
+   * 
+   *   A true value represents an exact match, while a false indicates 
+   *   a minimum number of columns required.
+   * 
+   * @return void
+   * 
+   * @throws \Exception
+   *  - The value 0 as number of columns.
+   */
+  public function setExpectedColumns($number_of_columns, $strict = FALSE) {
+
+    $context_key = $this->validator_context_key;
+
+    if ($number_of_columns <= 0) {
+      throw new \Exception('The setter method in ' . $context_key . ' requires an integer value greater than zero.');
```

Comment:
I think it's better to explicitly state which method since there is potential for future additional setter methods in this class.
```suggestion
      throw new \Exception('setExpectedColumns() requires an integer value greater than zero.');
```